src/griffe/docstrings/numpy.py

```diff
@@ -217,7 +217,9 @@ def _read_block(docstring: Docstring, *, offset: int) -> tuple[str, int]:
         |  # or
         (?P<name>{_RE_NAME})\s*:\s*  # just name
         |  # or
-        (?P<type>{_RE_TYPE})\s*  # just type
+        \s*:$  # no name, no type
+        |
+        :?\s*(?P<type>{_RE_TYPE})\s*  # just type
```

Comment:
Not tested, but this would look better to me:

```suggestion
        (?::\s*)?(?P<type>{_RE_TYPE})\s*  # just type
        |  # or
        :\s*  # no name, no type
```

The difference is: we don't allow leading spaces before the colon, but allow trailing spaces after it.

Reply:
Did a bit of tinkering just now, and I think because the `_RE_TYPE` regex is `.+`, the "just type" regex ends up capturing the "no name, no type" cases.

I think we may need the "no name, no type" regex first, since it's a subset of things our "just type" captures.

Reply:
Ha, sorry for making you do regex gymnastics! I thought my suggestion covered it, but if it's easier to have the  "nothing" case before the "just type" one, then please keep them in that order! I prefer the regex to stay simple and not individually cover badly-supported cases from composed ones. Each regex deals with its own issues :slightly_smiling_face: